libs/orders/src/lib/order-hooks/use-order-validation.tsx

```diff
@@ -28,33 +28,112 @@ export const useOrderValidation = ({
 }: ValidationProps) => {
   const { keypair } = useVegaWallet();
 
-  const invalidText = useMemo(() => {
+  const { message, isDisabled } = useMemo(() => {
     if (!keypair) {
-      return t('No public key selected');
+      return { message: t('No public key selected'), isDisabled: true };
     }
 
     if (keypair.tainted) {
-      return t('Selected public key has been tainted');
+      return {
+        isDisabled: true,
+        message: t('Selected public key has been tainted'),
+      };
+    }
+
+    if (
+      [
+        MarketState.Settled,
+        MarketState.Rejected,
+        MarketState.TradingTerminated,
+      ].includes(market.state)
+    ) {
+      return {
+        isDisabled: true,
+        message: t(
+          `This market is ${market.state.toLowerCase()} and no longer accepting orders`
```

Comment:
chore (non-blocking): we might want to add a map of market states to translations. Right now this could read "This market is tradingterminated and no longer accepting orders".

Perhaps a map like
```
const marketTranslations: { [S in MarketState]: string } = {
  [MarketState.TradingTerminated]: t('terminated'),
  [MarketState.Rejected]: t('rejected')
  ... // and so on
}
...
message: t(`This market is now ${marketTranslations[market.state]}' and is no longer accepting orders`)
```

Reply:
Worth checking the other resulting messages too.

Reply:
updated